src/main/java/edu/kit/kastel/dsis/mosim/refactorlizar/analysiscli/commands/EcoreTypeExtractionCommand.java

```diff
@@ -0,0 +1,76 @@
+package edu.kit.kastel.dsis.mosim.refactorlizar.analysiscli.commands;
+
+import static org.apache.logging.log4j.util.Strings.LINE_SEPARATOR;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.*;
+import java.util.*;
+import java.util.stream.Stream;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.shell.standard.ShellCommandGroup;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+@ShellComponent
+@ShellCommandGroup("Language Knowledge Extraction")
+public class EcoreTypeExtractionCommand {
+    private static final Logger logger = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+    private final String ECORE_FILE_ENDING = "ecore";
+    private final PathMatcher matcher =
+            FileSystems.getDefault().getPathMatcher("glob:**." + ECORE_FILE_ENDING);
+    private final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+    @ShellMethod("Metamodels Type Extraction")
+    public void showTypesInMetamodels(String rootPath) {
+        Map<String, List<String>> metamodels = new HashMap<>();
+        try (Stream<Path> paths = Files.walk(Paths.get(rootPath))) {
+            paths.filter(matcher::matches)
+                    .forEach(
+                            path -> {
+                                readMetamodels(metamodels, path);
```

Comment:
#### Ich mach mal hier die high level style warnung hin. Eigentlich ist eine Änderung hier nicht notwendig, aber vllt will man mal den Bloch Style gelesen haben.

Du könntest hier eigentlich einen schönen stream mit
```java
paths.
filter(matcher::matches)
.map(this::readMetamodels)
.forEach(this::printMetaModel);
```

readMetamodels ist ne java.util.Function von `Path -> Map<String,List<String>>`
und printMetaModel ein java.util.Consumer `von Map<String,List<String>>`
siehe effective java 43  bzw Kapitel 4. https://github.com/nicolasmanic/effective-java-in-a-nutshell/blob/master/README.md#item-43-prefer-method-references-to-lambdas